packages/redux/src/users/returns/actions/__tests__/fetchUserReturns.test.ts

```diff
@@ -0,0 +1,84 @@
+import * as actionTypes from '../../actionTypes';
+import { BlackoutError, getUserReturns } from '@farfetch/blackout-client';
+import { fetchUserReturns } from '../';
+import { INITIAL_STATE } from '../../../reducer';
+import {
+  mockNormalizedUserReturnsResponse,
+  mockUserReturnsResponse,
+  userId,
+} from 'tests/__fixtures__/users';
+import { mockStore } from '../../../../../tests';
+import find from 'lodash/find';
+
+jest.mock('@farfetch/blackout-client', () => ({
+  ...jest.requireActual('@farfetch/blackout-client'),
+  getUserReturns: jest.fn(),
+}));
+
+const usersMockStore = (state = {}) =>
+  mockStore({ users: INITIAL_STATE }, state);
+
+describe('fetchUserReturns action creator', () => {
+  let store = usersMockStore();
+  const query = {};
+  const expectedConfig = undefined;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    store = usersMockStore();
+  });
+
+  it('should create the correct actions for when the fetch user returns procedure fails', async () => {
+    const expectedError = new Error('fetch user returns error');
+
+    (getUserReturns as jest.Mock).mockRejectedValueOnce(expectedError);
+    expect.assertions(4);
```

Comment:
Instead of depending on `expect.assertions` which is error-prone, you can change the test a little bit to make it not depend on it like this:


```js
it('should create the correct actions for when the fetch user returns procedure fails', async () => {
    const expectedError = new Error('fetch user returns error');

    (getUserReturns as jest.Mock).mockRejectedValueOnce(expectedError);

    await expect(fetchUserReturns(userId, query)(store.dispatch)).rejects.toBe(
      expectedError,
    );
    expect(getUserReturns).toHaveBeenCalledTimes(1);
    expect(getUserReturns).toHaveBeenCalledWith(userId, query, expectedConfig);
    expect(store.getActions()).toEqual(
      expect.arrayContaining([
        { type: actionTypes.FETCH_USER_RETURNS_REQUEST },
        {
          type: actionTypes.FETCH_USER_RETURNS_FAILURE,
          payload: { error: expectedError },
        },
      ]),
    );
  });
```